### PR TITLE
Switch from enum to String for location and storageClass in BucketInfo

### LINKED ISSUE #354: Replace BucketInfo.StorageClass and BucketInfo.Location classes with plain strings.
Issue was raised by @mfschwartz and @thobrla which find the enum as a reference for the current
set of values confusing.

When replaced to strings we should add a link in the Javadoc of the relevant BucketInfo.Builder setter to a page with the list of values.

For location we can use - https://cloud.google.com/storage/docs/bucket-locations

For Storage class we can use - https://cloud.google.com/storage/docs/storage-classes
(and/or https://cloud.google.com/storage/docs/json_api/v1/buckets#resource as unfortunatly
the documentation of possible values is not as good as location - @mfschwartz is it possible to fix that?)

### PR DESCRIPTION
**Pull request by mziccard**
This PR fixes #354 and also adds javadoc to BucketInfo.

### CODE REVIEW

File: gcloud-java-storage/src/main/java/com/google/gcloud/storage/BucketInfo.java [hunk 1]
```diff
@@ -434,33 +361,57 @@ Builder selfLink(String selfLink) {
...
+    /**
+     * Sets whether blob versioning should be enabled or not for the bucket.
+     */
     public Builder versioningEnabled(Boolean enable) {
       this.versioningEnabled = firstNonNull(enable, Data.<Boolean>nullOf(Boolean.class));
       return this;
     }
 
+    /**
+     * Sets the bucket's website index page.
```
Comment (aozarov) [inline gcloud-java-storage/src/main/java/com/google/gcloud/storage/BucketInfo.java]: "Behaves as the bucket's directory index where missing objects are treated as potential directories."...
Similar to my suggestion in #356 I think we can improve on the comments of some of the setters/getters by using
a description similar to the ones described [here](https://cloud.google.com/storage/docs/json_api/v1/buckets#resource).

File: gcloud-java-storage/src/main/java/com/google/gcloud/storage/BucketInfo.java [hunk 2]
```diff
@@ -434,33 +361,57 @@ Builder selfLink(String selfLink) {
...
     public Builder deleteRules(Iterable<? extends DeleteRule> rules) {
       this.deleteRules = rules != null ? ImmutableList.copyOf(rules) : null;
       return this;
     }
 
-    public Builder storageClass(StorageClass storageClass) {
-      this.storageClass = firstNonNull(storageClass, StorageClass.NULL_VALUE);
+    /**
+     * Sets the bucket's storage class. A list of supported values is available
+     * <a href="https://cloud.google.com/storage/docs/json_api/v1/buckets#resource">here</a>.
```
Comment (aozarov) [inline gcloud-java-storage/src/main/java/com/google/gcloud/storage/BucketInfo.java]: Based on our comments the doc was updated to include the list of values in https://cloud.google.com/storage/docs/storage-classes (see "Name to use with APIs").

**Comment by mziccard** [comment]
I integrated BucketInfo javadoc

**Comment by aozarov** [comment]
Looks great! Just one nit.

**Comment by mziccard** [comment]
Fixed in place!

**Pull request merged**